IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -36,6 +37,43 @@
 static NSString *kWPJPrivateKeyIdentifier = @"com.microsoft.workplacejoin.privatekey\0";
 static NSString *kECPrivateKeyTagSuffix = @"-EC";
 
+@interface MSIDWPJMetadata : NSObject
+
+@property (nonatomic) NSString *certificateThumbprint;
+@property (nonatomic) NSString *cloudHost;
+@property (nonatomic) NSString *deviceID;
+@property (nonatomic) NSString *tenantIdentifier;
+@property (nonatomic) NSString *upn;
+
+- (NSDictionary *)serializeWithFormat:(BOOL)usePrimaryFormat;
+
+@end
+
+@implementation MSIDWPJMetadata
+
+- (NSDictionary *)serializeWithFormat:(BOOL)usePrimaryFormat
+{
+    NSMutableDictionary *result = [NSMutableDictionary new];
+    if (usePrimaryFormat)
+    {
+        result[MSID_PRIMARY_REGISTRATION_CERTIFICATE_THUMBPRINT] = self.certificateThumbprint;
+        result[MSID_PRIMARY_REGISTRATION_CLOUD] = self.cloudHost;
+        result[MSID_PRIMARY_REGISTRATION_DEVICE_ID] = self.deviceID;
+        result[MSID_PRIMARY_REGISTRATION_TENANT_ID] = self.tenantIdentifier;
+        result[MSID_PRIMARY_REGISTRATION_UPN] = self.upn;
+    }
+    else
+    {
+        result[MSID_DEVICE_INFORMATION_AAD_DEVICE_ID_KEY] = self.deviceID;
```

Comment:
for ecc registration, we dont send thumbprint and cloudHost ?

Reply:
For this case we send only 3 params (deviceID, upn, tenant). This format is being used by Defender iOS and they don't need other fields.  Also this is only when we read metadata locally, broker doesn't know about this "local" format.

Reply:
aah ok, did we close on Defender's dependency on certificateThumbprint ?. They were using it for comparing against the certificate thumbprint in graph.

Reply:
as far as I get, MacOS team needs thumbprint when iOS team doesn't. iOS defender uses this api in MSAL:
```

- (void)getWPJMetaDataDeviceWithParameters:(nullable MSALParameters *)parameters
                               forTenantId:(nullable NSString *)tenantId
                           completionBlock:(nonnull MSALWPJMetaDataCompletionBlock)completionBlock;
```

which returns in the local format and doesn't send request to the broker.

Defender for MacOS will be using this api:

```
- (void)getDeviceInformationWithParameters:(nullable MSALParameters *)parameters
                           completionBlock:(nonnull MSALDeviceInformationCompletionBlock)completionBlock;
```

it sends the request to broker and we return all data (including thumbprint) and the response in "primary/broker" format.

Reply:
aah sorry i missed this was only for ios. Thanks for explaining Serhii